docs/cookbook/promotions/custom-catalog-promotion-scope.rst

```diff
@@ -55,17 +54,16 @@ And the code for the provider itself:
             $this->productVariantRepository = $productVariantRepository;
         }
 
-        public function supports(CatalogPromotionScopeInterface $catalogPromotionScopeType): bool
-        {
-            return $catalogPromotionScopeType->getType() === self::TYPE;
-        }
-
-        public function provideEligibleVariants(CatalogPromotionScopeInterface $scope): array
+        public function inScope(CatalogPromotionScopeInterface $scope, ProductVariantInterface $productVariant): bool
         {
             $configuration = $scope->getConfiguration();
             Assert::keyExists($configuration, 'phrase', 'This rule should have configured phrase');
 
-            return $this->productVariantRepository->findByPhrase($configuration['phrase'], 'en_US');
+            return in_array(
+                $productVariant,
+                $this->productVariantRepository->findByPhrase($configuration['phrase'], 'en_US'),
+                true
+            );
```

Comment:
It will do almost similar thing(we will not check by code and english locale is not guaranteed), but lets not encourage to load additional data from db at this stage 

```suggestion
            return str_contains($productVariant->getName(), $configuration['phrase']);
```